std/typecons.d

```diff
@@ -6544,6 +6553,23 @@ pure @system unittest
     RefCounted!(int*) c;
 }
 
+@system unittest // Issue 20502
+{
+    import std.conv : to;
+    // Check that string conversion is transparent for refcounted
+    // structs that do not have either toString or alias this.
+    struct A { Object a; }
+    auto a  = A(new Object());
+    auto r = refCounted(a);
+    assert(to!string(r) == to!string(a));
+    assert(to!string(cast(const) r) == to!string(cast(const) a));
+    // Check that string conversion is still transparent for refcounted
+    // structs that have alias this.
+    struct B { int b; alias b this; }
+    struct C { B b; alias b this; }
+    assert(to!string(refCounted(C(B(123)))) == to!string(C(B(123))));
```

Comment:
Maybe add tests for `const/immutable RefCounted` (and maybe other storage classes as well)